File: cpp/leet_code/1614.cpp
// Optimizing the code for speed and unrolling loops for efficiency.
#pragma GCC optimize("O3,unroll-loops")
#include <string>

class Solution {
public:
    int maxDepth(std::string s) {
        int max_depth = 0;
        int depth = 0;
        for (int i = 0; i < s.size(); i++) {
            if (s[i] == '(') {
                depth++;
                max_depth = std::max(max_depth, depth);
            } else if (s[i] == ')') {
                depth--;
            }
        }
        return max_depth;
    }
};

File: csharp/leet_code/1614.cs
public class Solution {
    public int MaxDepth(string s) {
        int maxDepth = 0;
        int currentDepth = 0;
        foreach(char c in s) {
            if(c == '(') {
                currentDepth++;
                maxDepth = Math.Max(maxDepth, currentDepth);
            } else if(c == ')') {
                currentDepth--;
            }
        }
        return maxDepth;
    }
}